src/Goteo/Repository/DataSetRepository.php

```diff
@@ -0,0 +1,267 @@
+         }
+
+         $sql = "SELECT data_set.*
+                FROM data_set
+                INNER JOIN call_data_set cds ON cds.data_set_id = data_set.id
+                {$sqlWhere}
+                ORDER BY data_set.modified_at DESC
+                LIMIT 1
+                ";
+
+         $dataSet = $this->query($sql, $values)->fetchObject(DataSet::class );
+
+         if (!$dataSet instanceOf DataSet)
+             throw new ModelNotFoundException("DataSet not found");
+
+         return $dataSet;
+     }
+
+     public function persist(DataSet $dataSet, array &$errors = []): DataSet
+     {
+         if ($dataSet->getId())
+            return $this->update($dataSet, $errors);
+
+         return $this->create($dataSet, $errors);
+    }
+
+    private function create(DataSet $dataSet, &$errors = []): DataSet {
+        $fields = [
+            'title' => ':title',
+            'description' => ':description',
+            'type' => ':type',
+            'lang' => ':lang',
+            'url' => ':url',
+            'image' => ':image'
+        ];
+
+        $values = [
+            ':title' => $dataSet->getTitle(),
+            ':description' => $dataSet->getDescription(),
+            ':type' => $dataSet->getType(),
+            ':lang' => $dataSet->getLang(),
+            ':url' => $dataSet->getUrl(),
+            ':image' => $dataSet->getImage()->name
+        ];
+
+        $sql = "INSERT INTO `$this->table` (" . implode(',', array_keys($fields) ) . ") VALUES (" . implode(',', array_values($fields)) . ")";
+
+        try {
+            $this->query($sql, $values);
+            $dataSet->setId($this->insertId());
+        } catch (PDOException $exception) {
+            $errors[] = $exception->getMessage();
+            return false;
+        }
+        return $dataSet;
+
+    }
+
+    private function update(DataSet $dataSet, &$errors = []): DataSet {
+        $fields = [
+            'id' => ':id',
+            'title' => ':title',
+            'description' => ':description',
+            'type' => ':type',
+            'lang' => ':lang',
+            'url' => ':url',
+            'image' => ':image'
+        ];
+
+        $values = [
+            ':id' => $dataSet->getId(),
+            ':title' => $dataSet->getTitle(),
+            ':description' => $dataSet->getDescription(),
+            'type' => $dataSet->getType(),
+            ':lang' => $dataSet->getLang(),
+            ':url' => $dataSet->getUrl(),
+            ':image' => $dataSet->getImage()->name
+        ];
+
+        $sql = "REPLACE INTO `$this->table` (" . implode(',', array_keys($fields) ) . ") VALUES (" . implode(',', array_values($fields)) . ")";
+        try {
+            $this->query($sql, $values);
+        } catch (PDOException $exception) {
+            $errors[] = $exception->getMessage();
+        }
+        return $dataSet;
+    }
+
+    public function delete(DataSet $dataSet): void {
+        $sql = "DELETE FROM $this->table WHERE $this->table.id = :id";
```

Comment:
Aren't we missing the single quotes, as used in the `update()` method above. For both `$this->table` and `$this->table.id`?